contracts/eosio.system/src/wps.cpp

```diff
@@ -0,0 +1,758 @@
+        check(itr == _committees.end(), "This account has already been registered as a committee");
+
+        //add to the table
+        _committees.emplace(_self, [&](auto& committee){
+            committee.committeeman = committeeman;
+            committee.category = category;
+            committee.is_oversight = is_oversight;
+        });
+    }
+
+    void system_contract::edcommittee(name committeeman, const string& category, bool is_oversight){
+        //editing committee info requires contract account permissions
+        require_auth(get_self());
+
+        //verify that the committee account exists
+        check(is_account(committeeman), "committee account doesn't exist");
+
+        //verify that the size of the category string is not too long/short
+        check(category.size() > 0, "category should be more than 0 characters long");
+        check(category.size() < 64, "category should be less than 64 characters long");
+
+        auto itr = _committees.find(committeeman.value);
+        // verify that the account doesn't already exist in the table
+        check(itr != _committees.end(), "Account not found in committee table");
+
+        //add to the table
+        _committees.modify(itr, same_payer, [&](auto& committee){
+            committee.committeeman = committeeman;
+            committee.category = category;
+            committee.is_oversight = is_oversight;
+        });
+    }
+
+    void system_contract::rmvcommittee(name committeeman){
+        require_auth(_self);
+
+        check(is_account(committeeman), "committeeman account doesn't exist");
+
+        auto itr = _committees.find(committeeman.value);
+
+        check(itr != _committees.end(), "Account not found in committee table");
+        _committees.erase(itr);
+    }
+
+    void system_contract::rejectfund(name committeeman, name proposer, const string& reason){
+        require_auth(committeeman);
+
+        check(is_account(proposer), "Proposal creator does not exist");
+
+        check(reason.size() > 0, "must provide a brief reason");
+        check(reason.size() < 256, "reason is too long");
+
+        auto itr = _committees.find(committeeman.value);
+        // verify that the committee is on committee table
+        check(itr != _committees.end(), "Account not found in committee table");
+
+        auto itr_proposal = _proposals.find(proposer.value);
+        check(itr_proposal != _proposals.end(), "Proposal not found in proposal table");
+
+        check((*itr_proposal).committee == (*itr).committeeman || (*itr).is_oversight, "Committee is not associated with this proposal");
+        check((*itr_proposal).status == PROPOSAL_STATUS::APPROVED, "Proposal::status is not PROPOSAL_STATUS::APPROVED");
+
+        _proposals.modify(itr_proposal, same_payer, [&](auto& _proposal){
+            _proposal.status = PROPOSAL_STATUS::REJECTED;
+        });
+    }
+
+    double stake2vote2( int64_t staked ) {
```

Comment:
Any special reason to call it `stake2vote2()` and not `stake2vote()`.

Reply:
There's no practical difference between those two functions at the moment, but I could see the `stake2vote` algorithm differing slightly for WPS proposal voting compared to guild voting in the future. I separated the functions out so that we have more flexibility in the future if we want to make nuanced changes for staked voting in the two types of voting.